src/shogun/mathematics/graph/NGraph.cpp

```diff
@@ -10,12 +10,55 @@
 #include <shogun/mathematics/graph/runtime/ngraph/LogicalXor.h>
 #include <shogun/mathematics/graph/runtime/ngraph/Multiply.h>
 #include <shogun/mathematics/graph/runtime/ngraph/Subtract.h>
+#include <shogun/mathematics/graph/runtime/ngraph/MatMul.h>
+#include <shogun/mathematics/graph/runtime/ngraph/Dot.h>
+
 
 #include <ngraph/ngraph.hpp>
 
 using namespace shogun::graph;
 using namespace shogun::graph::detail::ngraph;
 
+
+inline int rowmajorindexfromcolumnmajorindex(int columnmajorindex, int width, int height) {
+    int row = columnmajorindex % height;
+    int column = columnmajorindex / height;
+    return row * width + column;
+}
+
+inline int columnmajorindexfromrowmajorindex(int rowmajorindex, int width, int height) {
+    return rowmajorindexfromcolumnmajorindex(rowmajorindex, height, width);
+}
+
+inline void column_to_row_major(void* dst, void* src, size_t type_size, size_t rows, size_t cols)
+{
+	size_t idx = 0;
+	for (size_t i = 0; i < rows; i++)
+	{
+		for (int j = 0; j < cols; ++j)
+		{
+			size_t new_idx = rowmajorindexfromcolumnmajorindex(idx, cols, rows);
+			memcpy((void*)&((char*)dst)[new_idx*type_size], (void*)&((char*)src)[idx*type_size], type_size);
+			idx++;
+		}
+	}	
+}
+
+inline void row_to_column_major(void* dst, void* src, size_t type_size, size_t rows, size_t cols)
+{
+	size_t idx = 0;
+	for (size_t i = 0; i < rows; i++)
+	{
+		for (int j = 0; j < cols; ++j)
+		{
+			size_t new_idx = columnmajorindexfromrowmajorindex(idx, cols, rows);
+			memcpy((void*)&((char*)dst)[new_idx*type_size], (void*)&((char*)src)[idx*type_size], type_size);
```

Comment:
@vigsterkr looks ugly but does the job... I can't be bothered to write a switch statement to cast to the correct type...
In any case we need to think more long term how to handle conversion from column to row major, this doesn't scale well.